packages/library/src/commonMain/kotlin/io/realm/Realm.kt

```diff
@@ -16,21 +16,34 @@
 package io.realm
 
 import io.realm.internal.RealmReference
+import io.realm.internal.SuspendableNotifier
 import io.realm.internal.SuspendableWriter
 import io.realm.internal.WeakReference
 import io.realm.internal.runBlocking
 import io.realm.interop.NativePointer
 import io.realm.interop.RealmInterop
 import kotlinx.atomicfu.AtomicRef
 import kotlinx.atomicfu.atomic
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
 // TODO API-PUBLIC Document platform specific internals (RealmInitilizer, etc.)
 class Realm private constructor(configuration: RealmConfiguration, dbPointer: NativePointer) :
     BaseRealm(configuration, dbPointer) {
 
-    private val writer: SuspendableWriter = SuspendableWriter(this)
+    internal val realmScope: CoroutineScope = CoroutineScope(SupervisorJob() + configuration.notificationDispatcher)
```

Comment:
True, this came from when I thought user actions would somehow also be using this. Although I suspect we will use this scope internally for the public callback API, so maybe better to just keep it?

Reply:
Why one job? can we use this supervisor to also cancel all observers when a Realm is closed?

Reply:
It is because Flows run inside user-defined scopes. I originally thought we might be able to interleave scopes, but that isn't possible.

But I suspect we will end up piping the change listener API through here. For which we don't want everything canceled if one fail.